### PR TITLE
[docs] Polish the first experience

### PR DESCRIPTION
**Pull request by oliviertassinari**
https://deploy-preview-261--material-ui-x.netlify.app/components/data-grid/

### CODE REVIEW

File: docs/public/_redirects [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Proxies
+# material-ui-x is configured to be hosted at the root.
+/static/* https://material-ui.netlify.app/static/:splat 200
```
Comment (oliviertassinari) [inline docs/public/_redirects:3]: should fix https://master--material-ui-x.netlify.app/components/data-grid/getting-started/#github

Capture d’écran 2020-09-11 à 00 18 40

File: docs/src/pages/components/data-grid/columns/CustomColumnTypesGrid.tsx [hunk 2]
```diff
@@ -27,22 +27,22 @@ const rows = [
...
-  valueFormatter: ({ value }) => valueFormatter.format(Number(value)),
+  valueFormatter: ({ value }) => currencyFormatter.format(Number(value)),
   cellClassName: 'font-tabular-nums',
 };
 
 const useStyles = makeStyles({
   root: {
     '& .font-tabular-nums': {
-      fontVariantNumeric: 'font-tabular-nums',
+      fontVariantNumeric: 'tabular-nums',
```
Comment (oliviertassinari) [inline docs/src/pages/components/data-grid/columns/CustomColumnTypesGrid.tsx:45]: Fix invalid CSS

File: docs/src/pages/components/data-grid/overview/DataGridDemo.tsx [hunk 3]
```diff
@@ -2,20 +2,21 @@ import * as React from 'react';
...
+    headerName: 'Age',
     type: 'number',
+    width: 90,
   },
   {
     field: 'fullName',
-    description: 'this column has a value getter and is not sortable',
-    headerClassName: 'highlight',
+    headerName: 'Full name',
+    description: 'This column has a value getter and is not sortable.',
```
Comment (oliviertassinari) [inline docs/src/pages/components/data-grid/overview/DataGridDemo.tsx:17]: Make sure the headers are clean: https://master--material-ui-x.netlify.app/components/data-grid/#mit-version.

File: packages/grid/data-grid/README.md [hunk 4]
```diff
@@ -14,13 +14,12 @@ npm install @material-ui/data-grid
...
 ```
 
-This component has 3 peer dependencies that you will need to install as well.
+This component has 2 peer dependencies that you will need to install as well.
 
 ```json
 "peerDependencies": {
   "@material-ui/core": "^4.9.12",
-  "react": "^16.8.0",
-  "styled-components": "^5.1.0"
```
Comment (oliviertassinari) [inline packages/grid/data-grid/README.md:23]: Forgotten in the previous changes

File: packages/grid/x-grid-data-generator/package.json [hunk 5]
```diff
@@ -16,6 +16,13 @@
   "bin": {
     "datagen": "./bin/data-gen-script.js"
   },
+  "scripts": {
```
Comment (oliviertassinari) [inline packages/grid/x-grid-data-generator/package.json:19]: first for consistency and ease of usage

File: packages/grid/x-grid-data-generator/src/commodities.columns.tsx [hunk 6]
```diff
@@ -32,20 +32,16 @@ import {
   randomTaxCode,
 } from './services';
 import {
-  CountryRenderer,
-  DoneRenderer,
-  EmailRenderer,
-  IncotermRenderer,
-  ProgressRenderer,
-  StatusRenderer,
+  renderCountry,
```
Comment (oliviertassinari) [inline packages/grid/x-grid-data-generator/src/commodities.columns.tsx:35]: I was confused by the upper case, I think they were components, turn out, they were, they were render functions. I have only realized it after trying using hooks inside, wasn't working.

File: packages/grid/x-grid-modules/src/components/checkbox-renderer.tsx [hunk 7]
```diff
@@ -28,16 +28,14 @@ export const HeaderCheckbox: React.FC<ColParams> = React.memo(({ api }) => {
...
-      <Checkbox
-        indeterminate={isIndeterminate}
-        checked={isChecked}
-        onChange={handleChange}
-        className="MuiDataGrid-checkboxInput"
-        color="primary"
-        inputProps={{ 'aria-label': 'Select All Rows checkbox' }}
-      />
-    </div>
+    <Checkbox
```
Comment (oliviertassinari) [inline packages/grid/x-grid-modules/src/components/checkbox-renderer.tsx:31]: lighter DOM structure. Made possible with display flex, align center, justify center. I assume the div was added to fix the non-circular checkbox.

File: packages/grid/x-grid-modules/src/components/styled-wrappers/GridRootStyles.ts [hunk 8]
```diff
@@ -174,8 +176,7 @@ export const useStyles = makeStyles(
         },
         '& .MuiDataGrid-cellWithRenderer': {
           display: 'flex',
-          flexDirection: 'column',
-          justifyContent: 'center',
+          alignItems: 'center',
```
Comment (oliviertassinari) [inline packages/grid/x-grid-modules/src/components/styled-wrappers/GridRootStyles.ts:179]: This change might be controversial. I was surprised by the previous behavior. I couldn't understand why my custom cell was full width;

File: packages/grid/x-grid-data-generator/src/renderer/renderLink.tsx [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+interface DemoLinkProps {
+  href: string;
+  children: string;
+}
+
+export const DemoLink = React.memo(function DemoLink(props: DemoLinkProps) {
+  const classes = useStyles();
+
+  const handleClick = (event) => {
+    event.preventDefault();
```
Comment (oliviertassinari) [inline packages/grid/x-grid-data-generator/src/renderer/renderLink.tsx:23]: Avoid opening links when testing the grid.

File: packages/grid/x-grid-data-generator/src/commodities.columns.tsx (hunk 6 above)
Comment (dtassone) [reply packages/grid/x-grid-data-generator/src/commodities.columns.tsx:35]: Yes it's true, the prop where they are attached is called `renderCell`

**Review by dtassone — APPROVED** [review]

**Pull request merged**